daemon/mgr/container_utils.go

```diff
@@ -251,19 +252,25 @@ func validateResource(r *types.Resources) ([]string, error) {
 
 	// validates memory cgroup value
 	if cgroupInfo.Memory != nil {
-		if r.Memory != 0 && !cgroupInfo.Memory.MemoryLimit {
+		if r.Memory > 0 && !cgroupInfo.Memory.MemoryLimit {
 			warn := "Current Kernel does not support memory limit, discard --memory"
 			logrus.Warn(warn)
 			warnings = append(warnings, warn)
 			r.Memory = 0
 			r.MemorySwap = 0
 		}
-		if r.MemorySwap != 0 && !cgroupInfo.Memory.MemorySwap {
+		if r.MemorySwap > 0 && !cgroupInfo.Memory.MemorySwap {
 			warn := "Current Kernel does not support memory swap, discard --memory-swap"
```

Comment:
Could we put the `warn` as the const or package var? WDYT?

Reply:
Do you mean every warn make a const or warn ? will this too many?